src/v/model/record_batch_reader.h

```diff
@@ -272,7 +272,58 @@ record_batch_reader
   make_memory_record_batch_reader(record_batch_reader::storage_t);
 
 record_batch_reader make_fragmented_memory_record_batch_reader(
-  fragmented_vector<model::record_batch>);
+  std::vector<record_batch_reader::storage_t> data);
+
+template<bool is_foreign, typename Container>
+std::vector<record_batch_reader::storage_t>
+make_fragmented_memory_storage_batches(Container batches) {
```

Comment:
We only have a few different containers that we'll use with this, so I'm not sure we need to move this into the header, and can use template function instantiation?

```cpp
// .h
template<bool is_foreign, typename Container>
std::vector<record_batch_reader::storage_t>
make_fragmented_memory_storage_batches(Container batches);

// .cc
// ... same as before ...
template
std::vector<record_batch_reader::storage_t>
make_fragmented_memory_storage_batches(chunked_vector);
```

Reply:
are you worried about build times ?

Reply:
yes